### PR TITLE
If Qt HiDpi scaling is active, use that insted of GRID_UNIT_PX

### PR DESCRIPTION
**Pull request by mariogrip**
This is the first move to depricate GRID_UNIT_PX all togheter

### CODE REVIEW

File: src/UbuntuToolkit/ucunits.cpp [hunk 1]
```diff
@@ -322,8 +326,16 @@ void UCUnits::windowPropertyChanged(QPlatformWindow *window, const QString &prop
...
+    // Update our pixel ratio to make sure it has not changed
+    m_devicePixelRatio = qGuiApp->devicePixelRatio();
+
     // choose integral grid unit value closest to requested scale
-    setGridUnit(qCeil(scale * DEFAULT_GRID_UNIT_PX) * m_devicePixelRatio);
+    // If qt hidpi scaling is enabled, use that as scaling
+    if (QHighDpiScaling::isActive())
+      setGridUnit(qCeil(m_devicePixelRatio * DEFAULT_GRID_UNIT_PX));
+    else
+      setGridUnit(qCeil(scale * DEFAULT_GRID_UNIT_PX));
```
Comment (mardy) [inline src/UbuntuToolkit/ucunits.cpp]: Shouldn't we keep this line as it was before, and multiply the value by `m_devicePixelRatio`?

Comment (mariogrip) [reply src/UbuntuToolkit/ucunits.cpp]: I though so too, but it seems to always return 1 if HighDpiScaling is disabled. I might be wrong here, but that's what i got from reading the qt's src.

Comment (mariogrip) [reply src/UbuntuToolkit/ucunits.cpp]: Looking at http://code.qt.io/cgit/qt/qtbase.git/tree/src/gui/kernel/qhighdpiscaling.cpp?h=5.9.6#n399 it seems to return 1

Comment (mardy) [reply src/UbuntuToolkit/ucunits.cpp]: Just to play 100% safe, I'd add it back, but it's your call :-)

Edit: OK, then, keep it as it is :-)

**Review by mardy — APPROVED** [review]

**Pull request merged**